src/XrmMockupShared/Utility.cs

```diff
@@ -546,25 +546,46 @@ public static Entity MakeProxyTypeEntity(Entity entity, Type ty)
             return (Entity)method.Invoke(entity, null);
         }
 
-        public static bool MatchesCriteria(DbRow row, FilterExpression criteria)
+        public static bool MatchesCriteria(Entity row, FilterExpression criteria)
         {
             if (criteria.FilterOperator == LogicalOperator.And)
-                return criteria.Filters.All(f => MatchesCriteria(row, f)) && criteria.Conditions.All(c => EvaluateCondition(row, c));
+                return criteria.Filters.All(f => 
+                    MatchesCriteria(row, f)) && 
+                    criteria.Conditions.All(c => EvaluateCondition(row, c));
             else
-                return criteria.Filters.Any(f => MatchesCriteria(row, f)) || criteria.Conditions.Any(c => EvaluateCondition(row, c));
+                return criteria.Filters.Any(f => 
+                    MatchesCriteria(row, f)) || 
+                    criteria.Conditions.Any(c => EvaluateCondition(row, c));
         }
 
 
-        private static bool EvaluateCondition(DbRow row, ConditionExpression condition)
+        private static bool EvaluateCondition(Entity row, ConditionExpression condition)
         {
-            if (condition.AttributeName == null)
+            object attr = null;
+            switch (condition)
             {
-                return Matches(row.Id, condition.Operator, condition.Values);
+                case var c when condition.AttributeName == null:
+                    return Matches(row.Id, condition.Operator, condition.Values);
+#if !XRM_MOCKUP_2011
+                case var c when condition.EntityName != null:
+                    var key = $"{condition.EntityName}.{condition.AttributeName}";
+                    if (row != null && row.Contains(key))
+                    {
+                        attr = row[key];
+                    }
+                    break;
+#endif
+                default:
+                    if (row.Contains(condition.AttributeName))
+                    {
+                        attr = row[condition.AttributeName];
+                    }
+                    break;
             }
 
-            var attr = row.GetColumn(condition.AttributeName);
-            if (attr is DbRow related) attr = related.Id;
-            var values = condition.Values.Select(v => ConvertToComparableObject(v));
+            attr = ConvertToComparableObject(attr);
+            if (attr is EntityReference entityRef) { attr = entityRef.Id; }
```

Comment:
As @Ladekarl  mentioned. This check is unnecessary as it is already handled in the ConvertToComparableObject method.